test/controllers/EventReportControllerSpec.scala

```diff
@@ -280,35 +280,95 @@ class EventReportControllerSpec extends AsyncWordSpec with Matchers with Mockito
     }
   }
 
-  "saveEvent" must {
-    "return 201 Created when valid response" in {
+  "getUserAnswersToCache" must {
+    "return 200 OK when valid response" in {
       val controller = application.injector.instanceOf[EventReportController]
 
-      when(mockEventReportService.saveEvent(
+      when(mockEventReportService.getEventFromMongo(
+        ArgumentMatchers.eq(pstr),
+        ArgumentMatchers.eq(Event1)
+      )(any()))
+        .thenReturn(Future.successful(Some(json)))
+
+      val result = controller.getUserAnswersFromCache(fakeRequest.withHeaders(
+        newHeaders = "pstr" -> pstr, "eventType" -> eventType))
+
+      status(result) mustBe OK
+    }
+
+    "return not found exception when invalid event type" in {
+      val controller = application.injector.instanceOf[EventReportController]
+
+      when(mockEventReportService.getEventFromMongo(
```

Comment:
This mock is useless here.
++ 
Adding mockEventReportService to reset and verify that this service never called because eventType is invalid - is truer I think. 
  ```
before {
    reset(mockAuthConnector, mockJSONPayloadSchemaValidator, mockEventReportService)
    when(mockAuthConnector.authorise[Option[String]](any(), any())(any(), any())) thenReturn Future.successful(Some("Ext-137d03b9-d807-4283-a254-fb6c30aceef1"))
    when(mockJSONPayloadSchemaValidator.validateJsonPayload(any(), any())) thenReturn Right(true)
  }
```
   ```
 "return not found exception when invalid event type" in {
      val controller = application.injector.instanceOf[EventReportController]

      recoverToExceptionIf[NotFoundException] {
        controller.getUserAnswersFromCache(fakeRequest.withHeaders(
          newHeaders = "pstr" -> pstr, "eventType" -> "test"))
      } map { response =>
        verify(mockEventReportService, never).getEventFromMongo(any(), any())(any())
        response.responseCode mustBe NOT_FOUND
        response.message must include("Bad Request: eventType (test) not found")
      }
    }
```

Reply:
we should also add one more test case for valid eventType and pstr but service will return None to us.

      when(mockEventReportService.getEventFromMongo(
        ArgumentMatchers.eq(pstr),
        ArgumentMatchers.eq(Event1)
      )(any()))
        .thenReturn(Future.successful(None))